### ISSUE TITLE
extend bytes error handling

### ISSUE THREAD

**Issue by stevesmoot** [zeek/spicy#1574]
in bytes::[lower|upper] can fail in undocumented ways #1251
(note it also applied to decode)

we ended up copying the Python method, or at least some of them.
Totally can work with those, but in fact would prefer a new one that may be more spicy-relevent
I'd call it QUIT, basically as soon as something doesnt work, return b""
The reason, is that this comes up in handling keywords in parsing (at least for me it does...) and if I have non-printable data then I know somethings wrong, I no longer care about matching it, and just seeing empty would be fine, it wont match a keyword.

So, if you think this makes sense, happy to try and add it as an option sometime.

**Comment by rsmmr** [comment]
Yeah, I can see that being useful, give it a try. Not quite sure about the name, could maybe see ABORT instead of QUIT?
